src/main/java/org/junit/runners/ParentRunner.java

```diff
@@ -346,8 +346,12 @@ protected final void runLeaf(Statement statement, Description description,
 
     @Override
     public Description getDescription() {
-        Description description = Description.createSuiteDescription(getName(),
-                getRunnerAnnotations());
+        Description description =
+                Description.createSuiteDescription(
+                        getTestClass().getJavaClass(),
```

Comment:
@kcooney is it work? If I set like test class null, method getName() return "null" how it will process next? But even if class can be null, code will work as they work before, because in that casen we initialize org.junit.runner.Description#fTestClass by null and in then if they null will try resolve by name 

org.junit.runner.Description#getTestClass
```java   
public Class<?> getTestClass() {
        if (fTestClass != null) {
            return fTestClass;
        }
        String name = getClassName();
        if (name == null) {
            return null;
        }
        try {
            fTestClass = Class.forName(name, false, getClass().getClassLoader());
            return fTestClass;
        } catch (ClassNotFoundException e) {
            return null;
        }
    }
```